### PR TITLE
[back] feat: add poll `presidentielle2022`

### PR DESCRIPTION
**Pull request by GresilleSiffle**
**related to** #601

---

This pull request adds the poll `pesidentielle2022` with its own criteria and a fixed list of entities (candidates).

I also found a bug in the `GET /entities/`, the URL parameter `poll_name` didn't filter the results.

Here is the list of criteria based on ours previous meetings:

```python
CRITERIA = [
    {
        "name": "be_president",
        "label_en": "Should be president",
        "label_fr": "Devrait être président.e",
        "optional": False,
    },
    {
        "name": "energy_environment",
        "label_en": "Energy & environment",
        "label_fr": "Énergie & environnement",
    },
    {
        "name": "education_culture",
        "label_en": "Education, culture & information",
        "label_fr": "Éducation, culture & information",
    },
    {
        "name": "health",
        "label_en": "Health & well-being",
        "label_fr": "Santé & bien-être",
    },
    {
        "name": "institutions_democracy",
        "label_en": "Institutions & democracy",
        "label_fr": "Institutions & démocratie",
    },
    {
        "name": "labour_economy",
        "label_en": "Labour & economy",
        "label_fr": "Travail & économie",
    },
    {
        "name": "solidarity",
        "label_en": "Solidarity & inclusivity",
        "label_fr": "Solidarité & inclusion",
    },
    {
        "name": "international",
        "label_en": "International politics",
        "label_fr": "Politique internationale",
    },
]
```

### CODE REVIEW

File: backend/tournesol/migrations/0033_add_poll_presidentielle2022.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    },
+    {
+        "name": "health",
+        "label_en": "Health & well-being",
+        "label_fr": "Santé et bien-être",
+    },
+    {
+        "name": "institutions_democracy",
+        "label_en": "Institutions & democracy",
+        "label_fr": "Institutions & démocracie",
```
Comment (amatissart) [inline backend/tournesol/migrations/0033_add_poll_presidentielle2022.py:48]: typo
```suggestion
        "label_fr": "Institutions & démocratie",
```

File: backend/tournesol/migrations/0033_add_poll_presidentielle2022.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    },
+    {
+        "name": "education_culture",
+        "label_en": "Education, culture & information",
+        "label_fr": "Éducation, culture & information",
+    },
+    {
+        "name": "health",
+        "label_en": "Health & well-being",
+        "label_fr": "Santé et bien-être",
```
Comment (amatissart) [inline backend/tournesol/migrations/0033_add_poll_presidentielle2022.py:43]: Why not "&" just like other criterias?

File: infra/ansible/inventory.yml [hunk 3]
```diff
@@ -126,6 +128,7 @@ all:
           django_database_name: tournesol
           django_database_user: tournesol
           django_email_backend: smtp
+          django_enable_api_wikidata: true
```
Comment (amatissart) [inline infra/ansible/inventory.yml:131]: I think this name can be misleading: even with `djano_enable_api_wikidata: false`, requests to Wikidata can be triggered from admin UI. To rename to show that it applies on migrations only, or update the implementation to apply this value both in migrations and on entity refresh.

Comment (GresilleSiffle) [reply infra/ansible/inventory.yml:131]: oops, I'll mention the migrations here.

**Review by lfaucon — APPROVED** [review]
Read the code and ran it locally. LGTM

**Comment by lfaucon** [comment]
I understood during Friday 18th standup that completing and deploying this PR would immediately make the presidentielle Poll visible in the frontend, but it doesn't seem to be the case as long as it is commented out in `frontend/src/utils/constants.ts`, isn't it?

**Comment by GresilleSiffle** [comment]
> I understood during Friday 18th standup that completing and deploying this PR would immediately make the presidentielle Poll visible in the frontend, but it doesn't seem to be the case as long as it is commented out in frontend/src/utils/constants.ts, is it?

@lfaucon if we merge this pull request, after the next deployment:

1. the `presidentielle2022` poll **won't be visible** in the front end
2. but the **API will be able to process incoming requests**

For me, it's safe to deploy because it's easy to wipe out the database in case of a malicious attack before officially launching this new poll.

That's also way I recommended to add a boolean field `is_active` on the Poll model to be able to enable and disable poll at will, making the back end able to reject requests coming requests to inactive polls. I still think it could be a nice addition in the future as it's easy to implement.

**Pull request merged**